frame/glutton/src/weights.rs

```diff
@@ -0,0 +1,172 @@
+// frame/glutton/src/weights.rs
+// --extra
+
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![allow(unused_parens)]
+#![allow(unused_imports)]
+
+use frame_support::{traits::Get, weights::{Weight, constants::RocksDbWeight}};
+use sp_std::marker::PhantomData;
+
+/// Weight functions needed for pallet_glutton.
+pub trait WeightInfo {
+	fn waste_ref_time_iter() -> Weight;
+	fn waste_proof_size_some() -> Weight;
+	fn waste_proof_size_none() -> Weight;
+	fn read_limits() -> Weight;
+	fn on_idle() -> Weight;
+	fn empty_on_idle() -> Weight;
+}
+
+/// Weights for pallet_glutton using the Substrate node and recommended hardware.
+pub struct SubstrateWeight<T>(PhantomData<T>);
+impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	fn waste_ref_time_iter() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 7_548_420 nanoseconds.
+		Weight::from_ref_time(7_696_066_000)
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	fn waste_proof_size_some() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `1521`
+		//  Estimated: `2499`
+		// Minimum execution time: 13_469 nanoseconds.
+		Weight::from_parts(14_549_000, 2499)
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	fn waste_proof_size_none() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `55`
+		//  Estimated: `2499`
+		// Minimum execution time: 3_061 nanoseconds.
+		Weight::from_parts(3_157_000, 2499)
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+	}
+	/// Storage: Glutton Compute (r:1 w:0)
+	/// Proof: Glutton Compute (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton Storage (r:1 w:0)
+	/// Proof: Glutton Storage (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	fn read_limits() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `55`
+		//  Estimated: `998`
+		// Minimum execution time: 3_316 nanoseconds.
+		Weight::from_parts(3_434_000, 998)
+			.saturating_add(T::DbWeight::get().reads(2_u64))
+	}
+	/// Storage: Glutton Storage (r:1 w:0)
+	/// Proof: Glutton Storage (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton Compute (r:1 w:0)
+	/// Proof: Glutton Compute (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton TrashData (r:2097 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(24), added: 2499, mode: MaxEncodedLen)
+	fn on_idle() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `124221`
+		//  Estimated: `5241401`
+		// Minimum execution time: 24_287_754 nanoseconds.
+		Weight::from_parts(24_680_302_000, 5241401)
```

Comment:
I dont get why this is so much lower. It should consume 100ms but only takes 24 ms :thinking:

Reply:
@ggwpez 
TL;DR You have to rerun the benchmark.

The reason for this is that during the benchmark `on_idle` read the weight for `waste_proof_size_some` from the  `weight.rs` file which contained the old weight(it is old because you have updated the constants in the meantime). 

So in essence the weight meter inside `on_idle` was not accurately metering the weight consumption since the `waste_at_most_ref_time` actually consumed less weight compared to before updating the constants.

https://github.com/paritytech/substrate/pull/12833/commits/5e622e589d7e35cb98f70ed24d217d99aa2aa25c#diff-708fa27eb3e02fea0f853aab6e6f5a773319cc0a6865e68edcda3d1ac3c357c0L235-R235